### PR TITLE
feat: Expose Optimizer, LR Scheduler and Checkpoint parameters in convenience functions

### LINKED ISSUE #353: Expose `ModelCheckpoint` parameters in convenience configuration functions
## Problem

If users want to set the parameters of the checkpoointing, they need to modify the values directly into the Pydantic configuration:

```python
from careamics.config import create_care_configuration

config = create_care_configuration(...)
config.training_config.checkpoint_callback.every_n_epochs = 5
```

We could instead allow a `checkpoint_kwargs`  or better yet the `ModelCheckpoint` Pydantic model in the API for the convenience functions.

### LINKED ISSUE #478: Expose `optimizer` and `lr_scheduler` parameters in convenience functions?
## Problem

Currently, `optimizer` and `lr_scheduler` can only be modified by modifying the configuration directly (or instantiating directly), since they are not exposed in `create_n2v_configuration` convenience function's signature:

https://github.com/CAREamics/careamics/blob/0fa617d7e27c61005274e433a3422be97bcadd7d/src/careamics/config/configuration_factories.py#L748-L767

We could add these parameters, as in some examples we do modify the LR and scheduler settings (see N2V BSD68)

### PR DESCRIPTION
**Pull request by jdeschamps**
## Description

> [!NOTE]  
> **tldr**: Allow users to pass optimizer (e.g., Adam), lr scheduler (e.g., ReduceLROnPlateau) and model checkpoint parameters in the convenience functions (e.g., `create_n2v_configuration`).

### Background - why do we need this PR?

Currently, advanced users wanting to change any of the optimizer or model checkpoint parameters (e.g., starting learning rate) must create a configuration and then in-place modify its parameters to get the behaviour they want.

NB: the parameter limits for the model checkpoint have also been updated to be less restrictive and adhere to the PyTorch Lightning documentation.

### Overview - what changed?

These advanced, yet important, parameters can now be passed directly to the convenience functions.

### Implementation - how did you implement the changes?

- The parameters were added to the convenience functions' signature
- `CheckpointModel`: the parameters limits have been updated to be more sensible.

### New features or files

- `optimizer` (name), `optimizer_params`, `lr_scheduler`, `lr_scheduler_params` and `checkpoint_params` have been added to functions in `configuration_factories.py`.

### Modified features or files

- `callback_model.py`: `CheckpointModel` parameters limit have been updated.

## How has this been tested?

Tests have been expanded.

## Related Issues

- Resolves #353 
- Resolves #478 
- Closes #381

## Breaking changes

No breaking change expected, possible merge conflicts with #476.

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

**Review by melisande-c — APPROVED** [review]
Looks good :)
